wcfsetup/install/files/lib/system/service/worker/Encryption.class.php

```diff
@@ -0,0 +1,187 @@
+    }
+
+    private static function addPadding(string $payload, string $contentEncoding): string
+    {
+        $length = Binary::safeStrlen($payload);
+        $paddingLength = ServiceWorkerHandler::MAX_PAYLOAD_LENGTH - $length;
+
+        if ($contentEncoding === ServiceWorker::CONTENT_ENCODING_AES128GCM) {
+            return \str_pad($payload . "\2", $paddingLength + $length, "\0", STR_PAD_RIGHT);
+        } elseif ($contentEncoding === ServiceWorker::CONTENT_ENCODING_AESGCM) {
+            return \pack('n*', $paddingLength) . \str_pad($payload, $paddingLength + $length, "\0", STR_PAD_LEFT);
+        } else {
+            throw new \RuntimeException('Unknown content encoding "' . $contentEncoding . '"');
+        }
+    }
+
+    private static function createInfo(string $type, ?string $context, string $contentEncoding)
+    {
+        if ($contentEncoding === ServiceWorker::CONTENT_ENCODING_AESGCM) {
+            \assert($context !== null);
+            \assert(Binary::safeStrlen($context) === 135);
+
+            return 'Content-Encoding: ' . $type . "\0" . Encryption::CURVE_ALGORITHM . $context;
+        } elseif ($contentEncoding === ServiceWorker::CONTENT_ENCODING_AES128GCM) {
+            return 'Content-Encoding: ' . $type . "\0";
+        } else {
+            throw new \RuntimeException('Unknown content encoding "' . $contentEncoding . '"');
+        }
+    }
+
+    private static function getEncryptionContentCodingHeader(
+        string $contentEncoding,
+        string $salt,
+        string $publicKey
+    ): string {
+        if ($contentEncoding === ServiceWorker::CONTENT_ENCODING_AES128GCM) {
+            return $salt
+                . \pack('N*', 4096)
+                . \pack('C*', Binary::safeStrlen($publicKey))
+                . $publicKey;
+        } elseif ($contentEncoding === ServiceWorker::CONTENT_ENCODING_AESGCM) {
+            return "";
+        } else {
+            throw new \RuntimeException('Unknown content encoding "' . $contentEncoding . '"');
+        }
+    }
+
+    /**
+     * {@link https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-encryption-encoding-00#section-4.2}
+     */
+    private static function createContext(
+        string $clientPublicKey,
+        string $serverPublicKey,
+        string $contentEncoding
+    ): ?string {
+        if ($contentEncoding === ServiceWorker::CONTENT_ENCODING_AES128GCM) {
+            return null;
+        }
+        \assert(Binary::safeStrlen($clientPublicKey) === 65);
+
+        $len = "\0A"; // 65 as Uint16BE
+
+        return "\0" . $len . $clientPublicKey . $len . $serverPublicKey;
+    }
+
+    private static function getIKM(
+        ServiceWorker $serviceWorker,
+        #[\SensitiveParameter] string $sharedSecret,
+        string $userPublicKey,
+        string $newPublicKey
+    ): string {
+        if ($serviceWorker->contentEncoding === ServiceWorker::CONTENT_ENCODING_AESGCM) {
+            $info = "Content-Encoding: auth\0";
```

Comment:
```suggestion
            $info = "Content-Encoding: auth\x00";
```
same below (and above).